miles/utils/eval_delegate.py

```
@@ -0,0 +1,158 @@
+            "arg_attrs": ("eval_max_response_len", "rollout_max_response_len"),
+        },
+    }
+
+    def __init__(self, args, dataset_cfg: Mapping[str, Any], defaults: Mapping[str, Any]):
+        self.name = str(dataset_cfg["name"])
+        defaults = defaults or {}
+        for field, spec in self.FIELD_SPECS.items():
+            dataset_value = _pick_from_mapping(dataset_cfg, spec["dataset_keys"])
+            default_value = _pick_from_mapping(defaults, spec["default_keys"])
+            arg_values = [getattr(args, attr, None) for attr in spec["arg_attrs"]]
+            setattr(self, field, _first_not_none(dataset_value, default_value, *arg_values))
+
+
+class EvalEnvConfig:
+    """Environment definition shared across delegate implementations."""
+
+    def __init__(self, args, raw_env_config: Mapping[str, Any], defaults: Mapping[str, Any]):
+        self.name = str(raw_env_config.get("name", "")).strip().lower()
+        self.url = raw_env_config.get("url")
+        self.timeout_secs = raw_env_config.get("timeout_secs", 3600)
+        self.max_retries = raw_env_config.get("max_retries", 1)
+        self.headers = dict(raw_env_config.get("headers", {}))
+        self.defaults = dict(defaults or {})
+
+
+def _rebuild_delegate_config(
+    args, raw_delegate_config: Sequence[Mapping[str, Any]], defaults: Mapping[str, Any]
+) -> List[EvalEnvConfig]:
+    envs: List[EvalEnvConfig] = []
+    defaults = defaults or {}
+    for env in raw_delegate_config or []:
+        env_name = str(env.get("name", "")).strip().lower()
+        if not env_name:
+            continue
+        if env_name == "skills":
+            from examples.eval.skills.skills_client import build_skills_eval_env_config
+
+            env_cfg = build_skills_eval_env_config(args, env, defaults)
+            if env_cfg is not None:
+                envs.append(env_cfg)
+        else:
+            logger.warning("Unknown delegate environment: %s", env_name)
```

nit: prefer fail fast